From 9fc4f31acbcce01d33f0b21df0485609047bcaea Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Thu, 30 Sep 2010 17:57:42 +0800
Subject: [PATCH] ENGR00132184-2 MX50: Port GPMI NAND driver to 2.6.35

Move __setup into driver, to remove the conflict between ARM2 and RDP.

Signed-off-by: Robby Cai <R63905@freescale.com>
---
 drivers/mtd/nand/gpmi-nfc/gpmi-nfc-main.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-main.c b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-main.c
index 2bd600e..be9e1a7 100644
--- a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-main.c
+++ b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-main.c
@@ -23,6 +23,8 @@
 #include "gpmi-nfc.h"
 #include "linux/slab.h"
 
+static int enable_gpmi_nand = { 0 };
+
 /*
  * This structure contains the "safe" GPMI timing that should succeed with any
  * NAND Flash device (although, with less-than-optimal performance).
@@ -1881,6 +1883,11 @@ static int __init gpmi_nfc_init(void)
 
 	pr_info("i.MX GPMI NFC\n");
 
+#ifdef	CONFIG_ARCH_MX50
+	if (enable_gpmi_nand == 0)
+		return 0;
+#endif
+
 	/* Register this driver with the platform management system. */
 
 	if (platform_driver_register(&gpmi_nfc_driver) != 0) {
@@ -1902,6 +1909,13 @@ static void __exit gpmi_nfc_exit(void)
 	platform_driver_unregister(&gpmi_nfc_driver);
 }
 
+static int __init gpmi_nand_setup(char *__unused)
+{
+	enable_gpmi_nand = 1;
+	return 1;
+}
+__setup("gpmi:nand", gpmi_nand_setup);
+
 module_init(gpmi_nfc_init);
 module_exit(gpmi_nfc_exit);
 
-- 
1.5.4.4

